ensure vpc-cni version is atleast 1.10.0 when ipv6 is configured

aclevername opened the pull request:
### Description

Closes #4463

## Problem
When version is not configured, we default to the EKS API to pick the latest version. For example:
```
2021-11-18 12:04:30 [ℹ]  describing addon versions for addon: vpc-cni
{
  Addons: [{
      AddonName: "vpc-cni",
      AddonVersions: [
        {
          AddonVersion: "v1.10.0-eksbuild.1",
          Architecture: ["amd64","arm64"],
          Compatibilities: [{
              ClusterVersion: "1.21",
              DefaultVersion: false,
              PlatformVersions: ["*"]
            }]
        },
        {
          AddonVersion: "v1.7.5-eksbuild.2",
          Architecture: ["amd64","arm64"],
          Compatibilities: [{
              ClusterVersion: "1.21",
              DefaultVersion: true,
              PlatformVersions: ["*"]
            }]
        },
```

atm `1.7.5` is the default, which wouldn't work for us. But this will probably change soon/before/with the ipv6 release if I was to guess. We can either:
1. Query the API, discover the default version, and then error if the default is not a good enough version. Adds a lot of complexity to the code, as well as doing API calls in the validation code, which we don't normally do AFAIK.
2. Do nothing. 

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

Himangini reviewed (approved):
LGTM 🎉

cPu1 commented on pkg/apis/eksctl.io/v1alpha5/validation.go:
```diff
@@ -184,6 +185,16 @@ func (c *ClusterConfig) ValidateVPCConfig() error {
+		}
+
+		if unsupportedVersion {
+			return fmt.Errorf("vpc-cni version must be at least version 1.10.0 for IPv6")
```
Just saw that Niki posted the same suggestion :slightly_smiling_face:

cPu1 commented:
>atm 1.7.5 is the default, which wouldn't work for us. But this will probably change soon/before/with the ipv6 release if I was to guess. We can either:

I also believe that it should change after the release. But in case this does not change after the IPv6 release, we should default to `1.10` or above if the default version in API is below `1.10`.

aclevername commented:
> > atm 1.7.5 is the default, which wouldn't work for us. But this will probably change soon/before/with the ipv6 release if I was to guess. We can either:
> 
> I also believe that it should change after the release. But in case this does not change after the IPv6 release, we should default to `1.10` or above if the default version in API is below `1.10`.

yeah agree'd thats the simplest solution. Implemented that and added issue to track https://github.com/weaveworks/eksctl/issues/4477
